### ISSUE TITLE
[FEATURE REQUEST] Integration with CalDav

### ISSUE THREAD

**Issue by madmath03** [jitsi/jitsi-meet#4719]
**Is your feature request related to a problem you are facing?**
Not related to a problem, more like something available with Jitsi Meet mobile App and would be nice to have in the Web (and Electron) App.

**Describe the solution you'd like**
Jitsi Meet currently allows you to connect to your Google or Microsoft Calendar.
On the other end, the (iPhone) Mobile App allows to access the phone's calendar which might be coming from anywhere (ie. not just Google or Microsoft).

I would like the Jitsi Meet app to allow me to connect to a CalDav calendar (for instance a private NextCloud server) and retrieve Jitsi Meeting URLs in my agenda.

**Describe alternatives you've considered**
None that I can think of so far.

**Comment by damencho** [comment]
I had spent some time on this when we were implementing calendar service and couldn't find some working js client-side API for doing this. Do you have anything in mind, that can work for a browser app? Any examples of web apps that do that without using external service?

**Comment by madmath03** [comment]
RoundCube has an [integration plugin](https://github.com/Offerel/roundcube_caldavzap) for [CalDavZAP](https://www.inf-it.com/open-source/clients/caldavzap/), "an open source CalDAV web client". I remember testing it and it seemed to work well with the NextCloud CalDAV server. Only issue I had in production was due to Cross Origin.
Not sure if CalDavZAP is still maintained though (last release being 2015...).

I know that RoundCube also has an external plugin at https://github.com/texxasrulez/Caldav_Calendar (based on https://plugins.roundcube.net/packages/kolab/calendar) but it's in PHP, not JS.

I will share anything I can find on the subject.

**Comment by madmath03** [comment]
Maybe https://github.com/lambdabaa/dav would be a better fit for Jitsi Meet but I have never used it.

**Comment by damencho** [comment]
So the CalDavZap and cdav-library are using incompatible licenses, so there are no go :(

Maybe the lambadabaa. I'm out of resources for the moment to look at it, as this requires and a working caldav server to test against, which I don't have ... I remember I was trying to install something so I can test, but was having problems with Cross Origin as you mentioned. 
But any help is welcome and I can review PRs :) 
The calendar integration is abstracted to 2-3 methods and is easy to implement, more work will be the UI for configuring this...

**Comment by mokkin** [comment]
If this is helping you I would be glad to offer you a CalDAV-Server (nextcloud) for testing purposes @damencho Just let me know...

**Comment by petervnv** [comment]
@damencho I can also provide you full access to a [radicale](https://github.com/Kozea/Radicale) server for testing if you'd like.

**Comment by damencho** [comment]
I don't think I will have the resources to work on this soon.

**Comment by resoli** [comment]
I can provide a NextCluod instance for testing.

**Comment by rasos** [comment]
`simple-caldav` is actively maintained and may be a starting point https://www.npmjs.com/package/simple-caldav

**Comment by Tiger862000** [comment]
I would like to push the request that was made by the OP and would like to generalize it as it was to specific IMHO...

**Is your feature request related to a problem you are facing?**
cannot connect my CalDav based calendar with Jit.si meet service.
 
**Describe the solution you'd like**
I would like connect my CalDav Nextcloud based calendar with the Jit.si meet service. Currently only closed / proprietary services are supported. 
As a user I would like:
1. ...to directly see my upcoming meeting in jitsi meet service and want to get notified
2. ...to offer scheduling link to my clients for them to schedule a meeting without email pingpong
3. ...to  retrieve Jitsi Meeting URLs in my agenda.

**Describe alternatives you've considered**
Cal.com is a scheduling platform like Calendly and offers Jitsi integration. They are open source, but in the free version offer limited features. This provides an alternative in my case, but it's a bit sad that an OSS app like Jitsi has to rely on a 3rd party for that.

**Possible solution**
Borrow code from the Cal.com CalDav integration https://github.com/calcom/cal.com

**Comment by saghul** [comment]
> Borrow code from the Cal.com CalDav integration https://github.com/calcom/cal.com

This is not an option because it's AGPL licensed.

There are other libraries like tsdav: https://github.com/natelindev/tsdav Someone just needs to do the work.

**Comment by shivamsaini124** [comment]
Hi @saghul,
Can you confirm if this issue has been resolved? If not, I would like to work on it.

**Comment by saghul** [comment]
This is the kind of work that's better suited for something like a GSoC project.

**Comment by shivamsaini124** [comment]
Thanks for the clarification! That makes sense, and it does look like an interesting problem to work on as part of a GSoC project.

**Comment by Avinash-Alapati** [comment]
Hi @saghul,
I’ve been exploring the calendar-sync architecture in Jitsi and looking into CalDAV client libraries like tsdav and simple-caldav.

From initial experiments it seems browser-side CalDAV access is likely to be blocked by CORS on most servers (e.g., Nextcloud), so a server-side proxy approach may be necessary for reliable web/electron support.

I’m currently prototyping a small Node-based CalDAV fetcher against a Nextcloud server to better understand auth methods and event parsing, and next I plan to map this to Jitsi’s calendar provider abstraction to see what preparatory work could be contributed.

If this aligns with expectations for a potential GSoC-sized effort, I’d appreciate any guidance on which parts of the codebase would be best to focus on first.

**Comment by damencho** [comment]
The integration is client based only, it should not require any backend.

**Comment by Avinash-Alapati** [comment]
Thanks for clarifying @damencho , I’ll focus on a client-only approach.
I’m currently testing CalDAV access directly from the browser using tsdav against a CORS-enabled Radicale instance to validate authentication and event fetching without any proxy.

Once I confirm feasibility, I’ll look into mapping this onto Jitsi’s existing calendar provider abstraction and share any findings or limitations I run into.

**Comment by jananivarthini455** [comment]
Hey @saghul @damencho 👋

I'm interested in working on this as part of my GSoC 2026 proposal. 
I've been contributing to jitsi-meet and would love to take this on.
Could you provide any guidance on where to start?
